Core/clim-basic/extended-streams/stream-input.lisp

```diff
@@ -319,224 +316,70 @@ keys read."))
+               (accelerator-gesture-p (gesture)
+                 (loop for gesture-name in *accelerator-gestures*
+                         thereis (event-matches-gesture-name-p gesture gesture-name))))
+          (cond
+            ((abort-gesture-p gesture)
+             (signal 'abort-gesture :event gesture))
+            ((accelerator-gesture-p gesture)
+             (signal 'accelerator-gesture :event gesture))
+            (t
+             (return-from stream-read-gesture gesture)))))))
+
+;;; FIXME STANDARD-EXTENDED-INPUT-STREAM isn't a character stream, however the
+;;; implementation of ACCEPT assumes it does.
 
 (defmethod stream-read-char-no-hang ((stream standard-extended-input-stream))
-  (with-encapsulating-stream (estream stream)
-    (loop
-      with char and reason
-      do (setf (values char reason) (stream-read-gesture estream :timeout 0))
-      until (or (characterp char) (eq reason :timeout) (eq reason :eof) )
-      finally (return (cond ((eq reason :timeout)
-                             nil)
-                            ((eq reason :eof)
-                             :eof)
-                            (t (char-for-read char)))))))
-
-(defmethod stream-unread-char ((stream standard-extended-input-stream)
-                               char)
-  (with-encapsulating-stream (estream stream)
-    (stream-unread-gesture estream char)))
-
-(defmethod stream-peek-char ((stream standard-extended-input-stream))
-  (with-encapsulating-stream (estream stream)
-    (loop
-      with char and reason
-      do (setf (values char reason) (stream-read-gesture estream :peek-p t))
-      until (or (characterp char) (eq reason :eof))
-      do (stream-read-gesture estream) ; consume pointer gesture
-      finally (return (if (eq reason :eof)
-                          reason
-                          (char-for-read char))))))
-
-(defmethod stream-listen ((stream standard-extended-input-stream))
-  (with-encapsulating-stream (estream stream)
-    (loop
-      with char and reason
-      do (setf (values char reason) (stream-read-gesture estream
-                                                         :timeout 0
-                                                         :peek-p t))
-      until (or (characterp char) (eq reason :eof) (eq reason :timeout))
-      do (stream-read-gesture estream) ; consume pointer gesture
-      finally (return (characterp char)))))
-
-(defmethod stream-clear-input ((stream standard-extended-input-stream))
-  (with-encapsulating-stream (estream stream)
-    (loop
-      with char and reason
-      do (setf (values char reason) (stream-read-gesture estream
-                                                         :timeout 0
-                                                         :peek-p t))
-      until (or (eq reason :eof) (eq reason :timeout))
-      do (stream-read-gesture estream) ; consume pointer gesture
-      ))
-  nil)
+  (loop
+    for ch = (stream-read-gesture stream)
```

Comment:
Shouldn't this specify `:timeout 0` like the method for `standard-input-stream`?

Reply:
it should, fixed